distribution.md

```diff
@@ -37,35 +37,82 @@ A `.cnb` file MUST be an uncompressed tar archive containing an OCI image. Its f
 ```
 
-A buildpack ID, buildpack version, and at least one stack MUST be provided in the OCI image config as a Label.
+A buildpack MUST contain a `buildpack.toml` file at its root directory.
+
+#### Labels
+
+For each buildpack layer, the buildpack ID and the buildpack version MUST be provided in `io.buildpacks.buildpack.layers`
 
-Label: `io.buildpacks.buildpackage.metadata`
+The following labels MUST be set in the buildpack image(through the image config's `Labels` field):
+
+| Label             | Description | 
+| --------          | -------- 
+| `io.buildpacks.buildpack.metadata`     | A JSON object representing Buildpack Metadata   |
+| `io.builpacks.buildpack.layers`| A JSON object representing the buildpack layers |
+
+
+`io.buildpacks.buildpackage.metadata` (JSON)
 ```json
 {
   "id": "<entrypoint buildpack ID>",
+  "name": "<buildpack name>",
   "version": "<entrypoint buildpack version>",
-  "stacks": [
+  "homepage": "<buildpack home page",
+  "targets": [
     {
-      "id": "<stack ID>",
-      "mixins": ["<mixin name>"]
+      "os": "<build-image os>",
+      "architecture": "<architecture>"
     }
   ]
 }
 ```
 
-The buildpack ID and version MUST match a buildpack provided by a layer blob.
-
-For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed stacks.
+`io.buildpacks.buildpack.layers` (JSON)
+```json
+{
+  "<buildpack ID>": {
+    "<buildpack version>": {
+      "api": "<buildpack API>",
+      "order": [
+        {
+          "group": [
+            {
+              "id": "<inner buildpack ID>",
+              "version": "<inner buildpack version>"
+            }
+          ]
+        }
+      ],
+      "layerDiffID": "<diff-ID>",
+      "homepage": "<buildpack homepage>",
+      "name": "<buildpack name>",
+    }
+  },
+  "<inner buildpack>": {
+    "<inner buildpack version>": {
+      "api": "<buildpack API>",
+      "targets" : [
+        {
+          "os": "<os buildpack supports>",
+          "architecture": "<underlying architecture>",
+        }
+      ],
+      "layerDiffID": "<diff-ID>",
+      "homepage": "<buildpack homepage>",
+      "name": "<buildpack name>",      
+    }
+  }
+}
+```
 
-For each listed stack, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
+The buildpack ID and version MUST match a buildpack provided by a layer blob.
 
-Each stack ID MUST only be present once.
-For a given stack, the `mixins` list MUST enumerate mixins such that no included buildpacks are missing a mixin for the stack.
+For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed targets.
 
-Fewer stack entries as well as additional mixins for a stack entry MAY be specified.
+For each listed target, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
```

Comment:
Not sure I understand this sentence to be honest...

Reply:
To be honest, even I am not a 100% about it, we had this in the earlier distribution spec and I chose not to remove it! I think that it pertains to the `io.buildpacks.buildpackage.metadata` label stating that for all the targets that we have in the targets field, all the buildpacks that work with that particular target must be a candidate for detection.

Reply:
@sclevine can we get some clarity here?